### PR TITLE
Contributing lib/ansible/modules/network/cloudengine/ce_vxlan_gateway.py module to manage HUAWEI data center CloudEngine

### PR DESCRIPTION
**Pull request by QijunPan**
##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

ce_vxlan_gateway.py
##### ANSIBLE VERSION

```
ansible 2.3.0
config file = /etc/ansible/ansible.cfg
configured module search path = ['/home/ansible-2.1.1.0/lib/ansible/modules/core/network/cloudengine', '/home/ansible-2.1.1.0/lib/ansible/module_utils', '/home/ansible-2.1.1.0/lib/ansible/modules', '/home/ansible-2.1.1.0', '/home/ansible-2.1.1.0/lib/ansible/utils/']
```

##### SUMMARY

```
TASK [create dfs-group] ********************************************************
task path: /usr1/code/OpenNESS/code/current/KVM/intg/test/testcases/test-ce_vxlan_gateway.yml:50
Using module file /usr/local/lib/python2.7/site-packages/ansible-2.3.0-py2.7.egg/ansible/modules/core/network/cloudengine/ce_vxlan_gateway.py
<ce_12800> ESTABLISH LOCAL CONNECTION FOR USER: root
<ce_12800> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1487054527.58-93814235766625 `" && echo ansible-tmp-1487054527.58-93814235766625="` echo $HOME/.ansible/tmp/ansible-tmp-1487054527.58-93814235766625 `" ) && sleep 0'
<ce_12800> PUT /tmp/tmpN1Vu8M TO /root/.ansible/tmp/ansible-tmp-1487054527.58-93814235766625/ce_vxlan_gateway.py
<ce_12800> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1487054527.58-93814235766625/ /root/.ansible/tmp/ansible-tmp-1487054527.58-93814235766625/ce_vxlan_gateway.py && sleep 0'
<ce_12800> EXEC /bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-tmp-1487054527.58-93814235766625/ce_vxlan_gateway.py; rm -rf "/root/.ansible/tmp/ansible-tmp-1487054527.58-93814235766625/" > /dev/null 2>&1 && sleep 0'
changed: [ce_12800] => {
    "changed": true, 
    "end_state": {
        "dfs_all_active": "disable", 
        "dfs_id": "1", 
        "dfs_source_ip": null, 
        "dfs_source_vpn": null, 
        "dfs_udp_port": null
    }, 
    "existing": {}, 
    "invocation": {
        "module_args": {
            "arp_direct_route": null, 
            "arp_distribute_gateway": null, 
            "auth_pass": null, 
            "authorize": false, 
            "dfs_all_active": null, 
            "dfs_id": "1", 
            "dfs_peer_ip": null, 
            "dfs_peer_vpn": null, 
            "dfs_source_ip": null, 
            "dfs_source_vpn": null, 
            "dfs_udp_port": null, 
            "host": "ce_12800", 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "port": 12346, 
            "provider": null, 
            "ssh_keyfile": null, 
            "state": "present", 
            "timeout": 10, 
            "transport": null, 
            "use_ssl": false, 
            "username": "root123", 
            "validate_certs": true, 
            "vbdif_bind_vpn": null, 
            "vbdif_mac": null, 
            "vbdif_name": null, 
            "vpn_instance": null, 
            "vpn_vni": null
        }, 
        "module_name": "ce_vxlan_gateway"
    }, 
    "proposed": {
        "dfs_all_active": null, 
        "dfs_id": "1", 
        "dfs_peer_ip": null, 
        "dfs_peer_vpn": null, 
        "dfs_source_ip": null, 
        "dfs_source_vpn": null, 
        "dfs_udp_port": null, 
        "state": "present"
    }, 
    "updates": [
        "dfs-group 1"
    ]
}
```

### CODE REVIEW

File: lib/ansible/modules/network/cloudengine/ce_vxlan_gateway.py [hunk 1]
```diff
@@ -0,0 +1,947 @@
...
+        else:
+            self.results['updates'] = list()
+
+        self.module.exit_json(**self.results)
+
+
+def main():
+    """Module main"""
+
+    argument_spec = dict(
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_vxlan_gateway.py:920]: For integers please:
use `type='int`
remove the `isdigit` check from `check_params`

File: lib/ansible/modules/network/cloudengine/ce_vxlan_gateway.py [hunk 2]
```diff
@@ -0,0 +1,947 @@
...
+            - Specifies the IPv4 address bound to a DFS group.
+              The value is in dotted decimal notation.
+        required: false
+        default: null
+    dfs_source_vpn:
+        description:
+            - Specifies the name of a VPN instance bound to a DFS group.
+              The value is a string of 1 to 31 case-sensitive characters without spaces.
+              If the character string is quoted by double quotation marks, the character string can contain spaces.
+              The value _public_ is reserved and cannot be used as the VPN instance name.
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_vxlan_gateway.py]: `The value C( _public_) is...`

File: lib/ansible/modules/network/cloudengine/ce_vxlan_gateway.py [hunk 3]
```diff
@@ -0,0 +1,947 @@
...
+            - Configure the IP address of an all-active gateway peer.
+              The value is in dotted decimal notation.
+        required: false
+        default: null
+    dfs_peer_vpn:
+        description:
+            - Specifies the name of the VPN instance that is associated with all-active gateway peer.
+              The value is a string of 1 to 31 case-sensitive characters, spaces not supported.
+              When double quotation marks are used around the string, spaces are allowed in the string.
+              The value _public_ is reserved and cannot be used as the VPN instance name.
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_vxlan_gateway.py]: `The value C( _public_) is...`

File: lib/ansible/modules/network/cloudengine/ce_vxlan_gateway.py [hunk 4]
```diff
@@ -0,0 +1,947 @@
...
+        required: false
+        default: null
+    vpn_instance:
+        description:
+            - Specifies the name of a VPN instance.
+              The value is a string of 1 to 31 case-sensitive characters, spaces not supported.
+              When double quotation marks are used around the string, spaces are allowed in the string.
+              The value _public_ is reserved and cannot be used as the VPN instance name.
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_vxlan_gateway.py]: `The value C( _public_) is...`

File: lib/ansible/modules/network/cloudengine/ce_vxlan_gateway.py [hunk 5]
```diff
@@ -0,0 +1,947 @@
...
+        description:
+            - Full name of VBDIF interface, i.e. Vbdif100.
+        required: false
+        default: null
+    vbdif_bind_vpn:
+        description:
+            - Specifies the name of the VPN instance that is associated with the interface.
+              The value is a string of 1 to 31 case-sensitive characters, spaces not supported.
+              When double quotation marks are used around the string, spaces are allowed in the string.
+              The value _public_ is reserved and cannot be used as the VPN instance name.
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_vxlan_gateway.py]: `The value C( _public_) is...`

File: lib/ansible/modules/network/cloudengine/ce_vxlan_gateway.py [hunk 6]
```diff
@@ -0,0 +1,947 @@
...
+        required: false
+        default: null
+    vbdif_mac:
+        description:
+            - Specifies a MAC address for a VBDIF interface.
+              The value is in the format of H-H-H. Each H is a 4-digit hexadecimal number, such as 00e0 or fc01.
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_vxlan_gateway.py]: `C(00e0) or C(fc01).`

File: lib/ansible/modules/network/cloudengine/ce_vxlan_gateway.py [hunk 7]
```diff
@@ -0,0 +1,947 @@
...
+              If an H contains less than four digits, 0s are added ahead. For example, e0 is equal to 00e0.
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_vxlan_gateway.py]: `C(e0) is equal to C(00e0)`

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/cloudengine/ce_vxlan_gateway.py (hunk 1 above)
Comment (QijunPan) [reply lib/ansible/modules/network/cloudengine/ce_vxlan_gateway.py:920]: Keep the string type to avoid the repeated conversion, thanks!

**Comment by QijunPan** [comment]
@gundalow   These problems have been fixed， can  this PR merged?

**Pull request merged**